File: DTOs/Enums/ContactType.swift
import Foundation

/// Перечисление, определяющее поддерживаемые типы контактных данных.
/// - `phone`: Телефонный номер.
/// - `email`: Электронная почта.
/// - `telegram`: Идентификатор Telegram.
/// - `whatsapp`: Идентификатор WhatsApp.
/// - `instagram`: Идентификатор Instagram.
public enum ContactType: String, Codable, CaseIterable, Hashable, Equatable {
    case phone
    case email
    case telegram
    case whatsapp
    case instagram
}

File: Android/shared/src/main/kotlin/shared/dto/enums/ContactType.kt
package shared.dto.enums

import kotlinx.serialization.SerialName

/// Описывает поддерживаемые типы контактных данных.
/// - `PHONE`    : Телефонный номер.
/// - `EMAIL`    : Электронная почта.
/// - `TELEGRAM` : Идентификатор Telegram.
/// - `WHATSAPP` : Идентификатор WhatsApp.
/// - `INSTAGRAM`: Идентификатор Instagram.
/// - `MESSAGE`  : Сообщение в приложении
enum class ContactType {
    @SerialName("phone")
    PHONE,

    @SerialName("email")
    EMAIL,

    @SerialName("telegram")
    TELEGRAM,

    @SerialName("whatsapp")
    WHATSAPP,

    @SerialName("instagram")
    INSTAGRAM,

    @SerialName("message")
    MESSAGE,
}
